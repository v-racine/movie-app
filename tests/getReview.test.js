const { Config } = require('../src/config/config');

Config.Get(process.env);

const AppFactory = require('../src/app');
const request = require('supertest');

describe('get a movie', () => {
  const mockPgClient = {};

  const app = AppFactory({
    pgClient: mockPgClient,
  });

  describe('when: users want to view ONE movie', () => {
    let rsp;
    let id = 3;

    let reviews = {
      rows: [
        {
          id: 3,
          reviewer: 'my chicken',
          grade: 'c',
          comments: 'stoopid chicken movie',
          movie_id: 4,
          movie_title: 'Point Break',
        },
      ],
    };

    beforeEach(async () => {
      mockPgClient.dbQuery = jest.fn().mockImplementation(() => {
        return new Promise((resolve) => {
          resolve(reviews);
        });
      });

      rsp = await request(app).get(`/reviews/${id}`).send();
    });

    afterEach(() => {
      jest.clearAllMocks();
    });

    test('then: we return all the information about that ONE review', async () => {
      expect(mockPgClient.dbQuery).toHaveBeenCalledWith(
        'SELECT * FROM reviews WHERE id = $1',
        id.toString(),
      );

      const text = rsp.text;
      expect(text).toMatchSnapshot();

      const status = rsp.status;
      expect(status).toBe(200);
    });
  });
});
