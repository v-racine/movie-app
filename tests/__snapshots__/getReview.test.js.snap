// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get a review when: a user requests a review that doesn't exist then: we return the error message AND display the view to create a new review 1`] = `
<!DOCTYPE html>
<html lang="en-US">
  <head>
    <title>
      The Ultimate Movie App
    </title>
    <meta charset="UTF-8">
    <link rel="stylesheet"
          href="/stylesheets/application.css"
    >
  </head>
  <body>
    <header>
      <h1>
        The Ultimate Movie App
      </h1>
    </header>
    <main>
      <p>
        Error: Review not found
      </p>
      <form action="/reviews/create"
            method="post"
      >
        <dl>
          <dt>
            <label for="title">
              Enter the title of a new movie:
            </label>
          </dt>
          <dd>
            <input type="text"
                   id="title"
                   name="title"
                   placeholder="title"
            >
          </dd>
          <dt>
            <label for="reviewer">
              Enter your name:
            </label>
          </dt>
          <dd>
            <input type="text"
                   id="reviewer"
                   name="reviewer"
                   placeholder="name"
            >
          </dd>
          <dt>
            <label for="grade">
              Enter your grade:
            </label>
          </dt>
          <dd>
            <select name="grade">
              <option value
                      disabled
                      selected
              >
                Choose one
              </option>
              <option value="A+">
                A+
              </option>
              <option value="A">
                A
              </option>
              <option value="A-">
                A-
              </option>
              <option value="B+">
                B+
              </option>
              <option value="B">
                B
              </option>
              <option value="B-">
                B-
              </option>
              <option value="C+">
                C+
              </option>
              <option value="C">
                C
              </option>
              <option value="C-">
                C-
              </option>
              <option value="D">
                D
              </option>
              <option value="F">
                F
              </option>
            </select>
          </dd>
          <dt>
            <label for="comments">
              What did you think of the movie?
            </label>
          </dt>
          <dd>
            <textarea id="comments"
                      name="comments"
                      rows="6"
                      cols="80"
                      placeholder="type your comments here"
            >
            </textarea>
          </dd>
        </dl>
        <fieldset class="actions">
          <input type="submit"
                 value="Submit Review"
          >
          <input type="reset"
                 value="Clear Form"
          >
        </fieldset>
      </form>
    </main>
  </body>
</html>
`;

exports[`get a review when: a user requests a review that doesn't exist when: users want to view ONE review then: we return all the information about that ONE review 1`] = `
<!DOCTYPE html>
<html lang="en-US">
  <head>
    <title>
      The Ultimate Movie App
    </title>
    <meta charset="UTF-8">
    <link rel="stylesheet"
          href="/stylesheets/application.css"
    >
  </head>
  <body>
    <header>
      <h1>
        The Ultimate Movie App
      </h1>
    </header>
    <main>
      <p>
        Here is your selected review:
        <table>
          <tbody>
            <tr>
              <td>
                <b>
                  Movie Title:
                </b>
                Point Break
              </td>
            </tr>
            <tr>
              <td>
                <b>
                  Reviewed by:
                </b>
                my chicken
              </td>
            </tr>
            <tr>
              <td>
                <b>
                  Grade:
                </b>
                c
              </td>
            </tr>
            <tr>
              <td>
                <b>
                  Comments:
                </b>
                stoopid chicken movie
              </td>
            </tr>
          </tbody>
        </table>
      </p>
      <footer>
        <form action="/reviews/update/3"
              method="get"
        >
          <button>
            Edit review
          </button>
        </form>
        <form action="/reviews/delete/3"
              method="post"
        >
          <button type="submit">
            Delete Review
          </button>
        </form>
      </footer>
    </main>
  </body>
</html>
`;
